Objects/typeobject.c

```diff
@@ -5963,21 +5963,31 @@ hackcheck(PyObject *self, setattrofunc func, const char *what)
     assert(PyTuple_Check(mro));
     Py_ssize_t i, n;
     n = PyTuple_GET_SIZE(mro);
+    int found_override = 0;
     for (i = 0; i < n; i++) {
         PyTypeObject *base = (PyTypeObject*) PyTuple_GET_ITEM(mro, i);
-        if (base->tp_setattro == func) {
-            /* 'func' is the earliest non-Python implementation in the MRO. */
+        if (base == &PyType_Type || base == &PyBaseObject_Type) {
```

Comment:
This is definitely not right. Special-casing types here (any type, really) will not help the general case.